### PR TITLE
feat(beat_html): render html_tailwind beats as a browser fragment

### PR DESCRIPTION
**Pull request by isamu**
## Summary

html_tailwind beat をブラウザ用 fragment にしました。**これで対応 beat 種別は 8/8** です（`#1526` の PR 13）。

- `src/utils/image_plugins/html_tailwind_markup.ts`（新規, pure）— 両経路が共有する規則
- `src/utils/beat_html/html_tailwind.ts`（新規）
- `dumpHtml` の `as` キャストを型ガードに置換

### 1. 設計判断は既に存在していました

script をどう扱うか悩みましたが、**Node の `dumpHtml` は既に script を出さず静的マークアップだけを返していました**（`elements` があれば `swipeElementsToHtml`、無ければ著者の `html`）。ブラウザ側も同じにするだけで、新しい判断は不要でした。

理由はブラウザ側でさらに強くなります — `innerHTML` 経由の `<script>` は実行されません。著者の `script`、swipe のアニメーション駆動、frame ベースの `animation` はいずれもスクリプト実行を要するので、**両経路とも「静止状態の beat」**を見せます。アニメーションだけの beat は、要素が初期位置で表示されます（何も出ないよりは preview として有用という判断）。

### 2. 著者の html はエスケープしません

この beat 種別では**マークアップが中身**で、スキーマは `script` も受け付けます。エスケープすると機能が消えるだけで、著者が持っていない能力を防ぐわけではありません。`BeatHtmlFragment.html` の「サニタイズしない」契約が最も効く箇所なので、そのことをコードのコメントに明記しました。

**ここは判断が要ります** — host 側でのサニタイズを前提にしてよいか、確認してください。

### 3. `as` キャストを型ガードに置き換えました

`dumpHtml` は `beat.image as { elements?: SwipeElement[] }` としていました。原因は `swipeElementSchema` が `z.lazy` の再帰のため `z.ZodType` と注釈され、`z.array(...)` が `unknown[]` になることです。

`SwipeElement` は**全フィールドが optional** なので、「各要素が非 null のオブジェクト」を確かめる `isSwipeElements` は**構造的に健全な narrowing** です（キャストの言い換えではありません）。zod は parse 時に既に形を検証しているので、これは型の穴を埋めるだけのものです — その旨をコメントに書いています。

### 4. Node 経路は「不変」ではありませんでした（round 1 で訂正）

当初「11 ケースで差分 0」と書きましたが、**生成器に配列以外の `elements` が無く**、Codex に指摘されました。旧コードの `Array.isArray` チェックを落としていたのが原因です。ガードに取り込んだうえで生成器を 17 ケースに広げ、throw も結果として比較したところ **2 件差分**が出ます:

| 入力 | main | 本PR |
|---|---|---|
| `elements: [null]` | **throw**（`Cannot read properties of null`） | html にフォールバック |
| `elements: ["str"]` | 文字列からゴミ markup を生成 | html にフォールバック |

**どちらも zod が parse 時に弾く形**なので実経路からは到達しません（5形すべて確認済み）。差分が出るのは parse を経ない呼び出しだけで、そこでは新挙動の方が良い（throw やゴミではなくフォールバック）ため、そのまま採用しテストで固定しました。

残る 15 ケースは差分 0 です。ハーネスが差分を見られることは変異（結合子 `\n`→`|`）でも確認しています。

### 5. id は2層で守られています

swipe element id は **parse 時（#1541 のスキーマ制約）** と **描画時（`assertSafeElementId`）** の両方で弾かれます。テストを書いていて、私が描画時だと思っていたものが実は parse 時に弾かれていたと分かったので、両方をテストに分けました。

## 検証

- `npx prettier --check` / `npx eslint`: clean
- `npx tsc --noEmit` / `npx tsc` / `cd types && npx tsc`: すべて clean
- `NODE_ENV=test npx tsx --test test/*/test_*.ts`: **1158 tests, 0 fail**
- browser-safety ゲート **7/7 pass**、バンドルは **38 入力 / 121.9 KB**（slide 時点から +3 / +2.8 KB）
- **変異3種すべてで赤くなることを確認**:

  | 変異 | 結果 |
  |---|---|
  | `elements` の優先を外す | fail=3 |
  | 配列の結合子を変える | fail=2 |
  | 空を `undefined` にしない | fail=1 |

## User Prompt

> b

Refs #1526

## Summary by CodeRabbit

* **New Features**
  * Added support for rendering `html_tailwind` content as browser-ready HTML fragments.
  * Supports raw HTML, multiple HTML entries, and swipe-based elements.
  * Produces static markup without scripts or animation-driven behavior.

* **Bug Fixes**
  * Improved consistency between fragment rendering and full document output.
  * Handles empty content and invalid element identifiers more reliably.

* **Tests**
  * Added coverage for rendering modes, script suppression, validation, and output consistency.

### CODE REVIEW

File: src/utils/image_plugins/html_tailwind_markup.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * schema also accepts a `script` — so escaping it would break the feature rather than
+ * protect anything the author does not already have.
+ */
+export type HtmlTailwindMarkupSource = {
+  html?: string | string[];
+  elements?: unknown[];
+};
+
+export const htmlTailwindMarkup = (image: HtmlTailwindMarkupSource): string => {
+  if (image.elements && image.elements.length > 0 && isSwipeElements(image.elements)) {
```
Comment (isamu) [inline src/utils/image_plugins/html_tailwind_markup.ts]: P2: This drops the old `Array.isArray(imageData.elements)` runtime guard, so an unparsed caller can now throw instead of falling back to `html`. Every site I found: `htmlTailwindMarkup` is called from `src/utils/image_plugins/html_tailwind.ts:255` and `src/utils/beat_html/html_tailwind.ts:21`; `isSwipeElements` is only called here. The old `dumpHtml` path handled `{ type: "html_tailwind", html: "<div>fallback</div>", elements: { length: 1 } }` by returning the fallback html because `Array.isArray(...)` was false. The new condition sees a truthy `.length` and then `isSwipeElements` calls `.every`, which throws (`TypeError: value.every is not a function`). This matters because the plugin/html path and the exported renderer can be called directly; the new tests even exercise an unparsed render-time path for bad IDs, so zod is not a universal precondition here. Smallest fix: make this branch `Array.isArray(image.elements) && image.elements.length > 0 && isSwipeElements(image.elements)` (or change `isSwipeElements` to accept `unknown` and perform that array check internally). What would change my mind: if the public contract is tightened so both `plugin.html` and `htmlTailwindToHtml` are only ever called after `mulmoHtmlTailwindMediaSchema.parse`, and tests for direct/unparsed render-time calls are removed or explicitly treated as unsupported.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED

Summary: I read every hunk in the PR diff against base `122415d5`. I found one P2 behavior-preservation/API-compatibility regression: the old `dumpHtml` path guarded `elements` with `Array.isArray`, while the new shared helper can throw on an unparsed non-array `elements` value with a truthy `length` instead of falling back to `html`. Inline: https://github.com/receptron/mulmocast-cli/pull/1548#discussion_r3835057264

Sandbox proof exit codes:

| Command | Exit code |
|---|---:|
| `NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts` | 0 |
| `npx tsc --noEmit` | 0 |
| `cd types && npx tsc && cd ..` | 0 |

Additional check: `npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts` without `NODE_ENV=test` exited 0.

Chromium has failed for me every round, so I did not imply or perform a browser check. I did not run a mutation sweep. Mutation I would name for the author to run after fixing: remove the restored `Array.isArray(image.elements)` guard and confirm a regression test for non-array `elements` fails.

| Axis | Result |
|---|---|
| correctness and edge cases | Finding: P2 non-array `elements` regression. Every site: `htmlTailwindMarkup` is called from `src/utils/image_plugins/html_tailwind.ts:255` and `src/utils/beat_html/html_tailwind.ts:21`; `isSwipeElements` is only called from `src/utils/image_plugins/html_tailwind_markup.ts:22`. Smallest fix: add `Array.isArray(image.elements)` before reading it as swipe elements, or make `isSwipeElements` accept `unknown` and check array-ness internally. What would change my mind: a documented contract that these functions are never called without prior schema parsing, plus removal/renaming of tests that currently assert render-time protection for unparsed callers. |
| C: THE TYPE GUARD | No P1 on required fields. I verified `SwipeElement` itself has all optional fields; nested `SwipeLoop.style` is required only inside optional `loop`, so an arbitrary non-null object is structurally assignable to `SwipeElement`. However, zod is not universal for all call paths: `plugin.html`/`htmlTailwindToHtml` can be called directly, and the PR’s own test calls the renderer without parsing for bad IDs. |
| C: BEHAVIOUR PRESERVATION | Finding covered above. The generator missed an input shape old code handled: `{ type: "html_tailwind", html: "<div>fallback</div>", elements: { length: 1 } }`. Old code returned fallback html because `Array.isArray` was false; new code throws because `.every` is missing. Yes, removal of `Array.isArray(imageData.elements)` is a real behavior difference for unparsed inputs. |
| C: THE NO-ESCAPING DECISION | No finding. I verified raw author html was already the Node dump behavior (`joinHtml(imageData.html ?? "")`) and `mulmoHtmlTailwindMediaSchema` accepts `script`. Escaping here would remove intended markup, not remove a capability this beat type lacks. The host contract is stated clearly enough in `BeatHtmlFragment.html` and reinforced in the new helper comments: not sanitized, sanitize before DOM insertion. |
| C: SCRIPT OMISSION | No finding. I verified old `dumpHtml` emitted swipe markup or author html only; it did not emit author `script`, swipe animation script, or frame animation. Returning rest-state elements for animation-only swipe content is consistent with the existing dump/list/editor-preview behavior. A script-only beat returns `undefined`, which is also covered. |
| security | No new finding beyond the API-compat issue. Raw html remains an explicit unsanitized fragment contract; hosts remain responsible for sanitization before DOM insertion. Safe element IDs still have parse-time and render-time checks. |
| tests | No finding. Required targeted tests passed with `NODE_ENV=test`; new targeted tests also passed without `NODE_ENV=test`. The missing test is for malformed/unparsed non-array `elements` preserving the old fallback behavior. |
| API compatibility | Finding: the direct renderer/plugin API is less tolerant of malformed `elements` than before. See inline P2. |
| consistency | No finding. Browser fragment and Node dump now share one helper, which reduces drift once the array guard is restored. Supported beat dispatch/list tests were updated consistently. |
| i18n-a11y | No findings. I did not see changed user-facing text, locale-sensitive formatting, or accessibility semantics beyond raw user-provided markup. |
| browser safety | No finding from the automated gate. The targeted browser-safety test passed and reports the bundle path at 38 inputs / 121.9kb per the PR context. Chromium has failed for me every round, so no browser runtime measurement was performed by me. |

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.

**Comment by isamu** [comment]
## Codex exchange — iteration 1

<details><summary>Prompt</summary>

```text
Review PR #1548 at https://github.com/receptron/mulmocast-cli/pull/1548.
Repo: /Users/isamu/ss/llm/mulmocast-cli. Base is main at 122415d5.

── BEFORE YOU REVIEW: prove your sandbox works ──────────────────────────────
Report each EXIT CODE verbatim:
  1. NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts
  2. npx tsc --noEmit
  3. cd types && npx tsc && cd ..
Chromium has failed for you every round; say so rather than implying a browser check.
DO NOT run a mutation sweep. NAME any mutation and I will run it.

── THIS IS THE ONLY ROUND YOU ARE GUARANTEED ────────────────────────────────
Read the WHOLE diff before posting. Post inline via
`gh api repos/receptron/mulmocast-cli/pulls/1548/comments`, top-level via `gh pr comment 1548`.

── TIER C — a cast replaced by a guard, a behaviour-preservation claim, and a
   deliberate no-escaping decision on user markup ──────────────────────────
── AXES — answer EVERY line, even when the answer is 'no findings' ──────────
  correctness and edge cases
  C: THE TYPE GUARD — `isSwipeElements` narrows `unknown[]` to `SwipeElement[]` by checking
     each entry is a non-null object. The argument is that EVERY SwipeElement field is
     optional, so any object satisfies it structurally. VERIFY that claim against the
     interface — if even one field is required, the guard is a cast wearing a guard's
     clothes and that is a P1. Also: is it right that zod has already validated the shape,
     or is there a call path that reaches this without parsing?
  C: BEHAVIOUR PRESERVATION — dumpHtml is claimed unchanged, measured over 11 inputs with a
     mutation producing 1 difference. Attack the generator: name an input shape it does not
     produce where old and new could differ. Note the old code had
     `Array.isArray(imageData.elements)` and the new code does not — is that a difference?
  C: THE NO-ESCAPING DECISION — the author's html is emitted raw. The argument is that this
     beat type accepts a `script` anyway so escaping removes a feature rather than a
     capability. Is that right, and is the contract stated clearly enough for a host?
  C: SCRIPT OMISSION — `script`, swipe animation and `animation: true` all produce nothing.
     A beat whose only content is an animation renders its elements at rest. Is that the
     right call for a list/editor preview, or should it return undefined instead?
  security / tests / API compatibility / consistency / i18n-a11y / browser safety
     (bundle went to 38 inputs / 121.9kb)

An axis with no findings is a RESULT and has to be stated.

── EACH FINDING CARRIES FOUR THINGS ─────────────────────────────────────────
1. SEVERITY — P1 / P2 / P3.  2. EVERY SITE (grep).  3. THE SMALLEST CHANGE THAT RESOLVES IT.
4. WHAT WOULD CHANGE YOUR MIND.

── CONTEXT ──────────────────────────────────────────────────────────────────
- Tests: `NODE_ENV=test npx tsx --test ./test/*/test_*.ts`; new tests must also pass without
  NODE_ENV=test. tsconfig `include` is ['src'] so tsc does not typecheck test/.
- eslint-disable / @ts-ignore / `as` casts are forbidden by the repo's rules.
- Only the index.browser.ts export remains in #1526 after this.
- Across this series several of MY claims were wrong. Treat every claim in this PR body as
  suspect and say which you verified rather than read.

── ALREADY SETTLED ──────────────────────────────────────────────────────────
| # | iter | finding | disposition | why |
|---|------|---------|-------------|-----|
| 0 | setup | Chromium 起動不可。ブラウザ実測はすべて当方。 | OK | |

── END WITH ONE TOP-LEVEL COMMENT ───────────────────────────────────────────
  'CODEX VERDICT: LGTM' or 'CODEX VERDICT: CHANGES REQUESTED' + summary, then the axis
  table, then on its own line:
    'FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.'
Do not apply any fixes yourself.
```

</details>

<details><summary>Codex reply</summary>

```text
Reviewed PR #1548 and posted:

Inline finding:
https://github.com/receptron/mulmocast-cli/pull/1548#discussion_r3835057264

Top-level verdict:
https://github.com/receptron/mulmocast-cli/pull/1548#issuecomment-5377815345

Required exit codes reported verbatim in the PR comment:

1. `NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts`: `0`
2. `npx tsc --noEmit`: `0`
3. `cd types && npx tsc && cd ..`: `0`

Verdict was `CODEX VERDICT: CHANGES REQUESTED` for the missing `Array.isArray` runtime guard behavior regression. I also stated that Chromium has failed for me every round and that I did not perform a browser check.
```

</details>

### Triage — round 1

| finding | disposition |
|---|---|
| `Array.isArray` を落としたことによる挙動差 | **FIXED** `48e34a21` |

**私が「攻撃してくれ」と書いた箇所そのものを突かれました。** 共有モジュールに移す際に旧コードの `Array.isArray` を落としており、生成器が配列しか作っていなかったので「差分 0」と出ていました。

`Array.isArray` をガードに取り込み、生成器を 17 ケースに広げ、**throw も結果として比較**したところ 2 件差分:

| 入力 | main | 本PR |
|---|---|---|
| `elements: [null]` | **throw** | html にフォールバック |
| `elements: ["str"]` | 文字列からゴミ markup | html にフォールバック |

5形すべて zod が parse 時に弾くので**実経路からは到達しません**。差分が出るのは parse を経ない呼び出しだけで、そこでは新挙動の方が良いため採用し、テストで固定しました。

「0 という数字は生成器についての主張でしかない」という教訓が、このシリーズで何度目かの形で出ています。

**Comment by isamu** [comment]
## Codex exchange — iteration 2

<details><summary>Prompt</summary>

```text
Review PR #1548 (round 2) at https://github.com/receptron/mulmocast-cli/pull/1548.
Repo: /Users/isamu/ss/llm/mulmocast-cli.

── SANDBOX ──────────────────────────────────────────────────────────────────
Report EXIT CODEs for:
  1. NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts
  2. npx tsc --noEmit
  3. cd types && npx tsc && cd ..
Chromium has failed for you every round; say so rather than implying a browser check.

── WHAT CHANGED SINCE ROUND 1 ───────────────────────────────────────────────
Head is 48e34a21. You were right, and it was the exact gap I had asked you to attack: I
dropped `Array.isArray` when moving the rule into the shared module, and my generator only
built arrays, so the harness reported 0 over a shape it never produced.

Array.isArray now lives inside isSwipeElements. Re-measured with the missing shapes and with
throws compared as outcomes — 17 inputs, **2 differences**:

    elements: [null]   main: threw "Cannot read properties of null"  ->  new: falls back to html
    elements: ["str"]  main: rendered garbage from a string          ->  new: falls back to html

Both shapes are rejected by zod at parse time (verified for [null], ["str"], "not-an-array",
7 and {}), so neither is reachable through the pipeline, and in both the new behaviour is
better. Kept deliberately and pinned by a test; the PR body now states 2 differences rather
than 0.

Review this head:
  - Is keeping those two differences right, or should the fragment reproduce main's throw?
  - Is `isSwipeElements` now correct — does `Array.isArray(value) && value.every(...)`
    narrow soundly, and is the "every SwipeElement field is optional" claim true? You did
    not say whether you verified that in round 1.
  - Is the generator now complete, or is there still a shape it does not build?
  - Anything round 1's change invalidated.

── ALREADY SETTLED ──────────────────────────────────────────────────────────
| # | iter | finding | disposition | why |
|---|------|---------|-------------|-----|
| 0 | setup | Chromium 起動不可。ブラウザ実測はすべて当方。 | OK | |
| 1 | 1 | Array.isArray 欠落による挙動差 | FIXED 48e34a21 | ガードに取り込み、17ケースで再測定、2差分を意図的に採用 |

── END WITH ONE TOP-LEVEL COMMENT ───────────────────────────────────────────
  'CODEX VERDICT: LGTM' or 'CODEX VERDICT: CHANGES REQUESTED' + summary, then the axis
  table, then on its own line:
    'FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.'
Do not apply any fixes yourself.
```

</details>

<details><summary>Codex reply</summary>

```text
CODEX VERDICT: LGTM

No blocking findings. Keeping the two intentional differences is right: `[null]` and `["str"]` are zod-rejected before the normal pipeline, and fallback is better than reproducing main’s throw/garbage for bypass callers.

| Axis | Result | Notes |
|---|---:|---|
| Head reviewed | OK | `48e34a2177ebabd29ec52635339313f38a5260d5` |
| `isSwipeElements` | OK | `Array.isArray(value) && value.every(non-null object)` is sound for `SwipeElement[]` structurally; I verified the `SwipeElement` top-level fields are all optional. Nested types can have required fields, but those are protected by zod on the real path. |
| Generator coverage | OK | Complete for the relevant branch shapes: missing/non-array/empty array/valid object entries/invalid primitive entries. It is not an exhaustive fuzzer for arbitrary zod-bypassed malformed nested objects, but that does not expose a reachable pipeline gap. |
| Round-1 invalidation | None found | The shared guard restores the old array gate and improves only unreachable bypass cases. |
| Browser check | Not run | Chromium has failed every round, so I am not implying a browser check. |
| Requested test 1 | exit code `0` | `NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts` |
| Requested test 2 | exit code `0` | `npx tsc --noEmit` |
| Requested test 3 | exit code `0` | `cd types && npx tsc && cd ..` |

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.
```

</details>

**Comment by isamu** [comment]
## Codex exchange — iteration 2

⚠️ **Codex はこのラウンドの verdict を PR に投稿できていません**（stdout には出力済み、GitHub への post が行われず）。verdict とテーブルは以下に verbatim で中継します。

<details><summary>Prompt</summary>

```text
Review PR #1548 (round 2) at https://github.com/receptron/mulmocast-cli/pull/1548.
Repo: /Users/isamu/ss/llm/mulmocast-cli.

── SANDBOX ──────────────────────────────────────────────────────────────────
Report EXIT CODEs for:
  1. NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts
  2. npx tsc --noEmit
  3. cd types && npx tsc && cd ..
Chromium has failed for you every round; say so rather than implying a browser check.

── WHAT CHANGED SINCE ROUND 1 ───────────────────────────────────────────────
Head is 48e34a21. You were right, and it was the exact gap I had asked you to attack: I
dropped `Array.isArray` when moving the rule into the shared module, and my generator only
built arrays, so the harness reported 0 over a shape it never produced.

Array.isArray now lives inside isSwipeElements. Re-measured with the missing shapes and with
throws compared as outcomes — 17 inputs, **2 differences**:

    elements: [null]   main: threw "Cannot read properties of null"  ->  new: falls back to html
    elements: ["str"]  main: rendered garbage from a string          ->  new: falls back to html

Both shapes are rejected by zod at parse time (verified for [null], ["str"], "not-an-array",
7 and {}), so neither is reachable through the pipeline, and in both the new behaviour is
better. Kept deliberately and pinned by a test; the PR body now states 2 differences rather
than 0.

Review this head:
  - Is keeping those two differences right, or should the fragment reproduce main's throw?
  - Is `isSwipeElements` now correct — does `Array.isArray(value) && value.every(...)`
    narrow soundly, and is the "every SwipeElement field is optional" claim true? You did
    not say whether you verified that in round 1.
  - Is the generator now complete, or is there still a shape it does not build?
  - Anything round 1's change invalidated.

── ALREADY SETTLED ──────────────────────────────────────────────────────────
| # | iter | finding | disposition | why |
|---|------|---------|-------------|-----|
| 0 | setup | Chromium 起動不可。ブラウザ実測はすべて当方。 | OK | |
| 1 | 1 | Array.isArray 欠落による挙動差 | FIXED 48e34a21 | ガードに取り込み、17ケースで再測定、2差分を意図的に採用 |

── END WITH ONE TOP-LEVEL COMMENT ───────────────────────────────────────────
  'CODEX VERDICT: LGTM' or 'CODEX VERDICT: CHANGES REQUESTED' + summary, then the axis
  table, then on its own line:
    'FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.'
Do not apply any fixes yourself.
```

</details>

<details><summary>Codex reply（verbatim stdout — 未投稿のため中継）</summary>

```text
CODEX VERDICT: LGTM

No blocking findings. Keeping the two intentional differences is right: `[null]` and `["str"]` are zod-rejected before the normal pipeline, and fallback is better than reproducing main’s throw/garbage for bypass callers.

| Axis | Result | Notes |
|---|---:|---|
| Head reviewed | OK | `48e34a2177ebabd29ec52635339313f38a5260d5` |
| `isSwipeElements` | OK | `Array.isArray(value) && value.every(non-null object)` is sound for `SwipeElement[]` structurally; I verified the `SwipeElement` top-level fields are all optional. Nested types can have required fields, but those are protected by zod on the real path. |
| Generator coverage | OK | Complete for the relevant branch shapes: missing/non-array/empty array/valid object entries/invalid primitive entries. It is not an exhaustive fuzzer for arbitrary zod-bypassed malformed nested objects, but that does not expose a reachable pipeline gap. |
| Round-1 invalidation | None found | The shared guard restores the old array gate and improves only unreachable bypass cases. |
| Browser check | Not run | Chromium has failed every round, so I am not implying a browser check. |
| Requested test 1 | exit code `0` | `NODE_ENV=test npx tsx --test test/beat_html/test_html_tailwind.ts test/beat_html/test_browser_safety.ts` |
| Requested test 2 | exit code `0` | `npx tsc --noEmit` |
| Requested test 3 | exit code `0` | `cd types && npx tsc && cd ..` |

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.
```

</details>

### Triage — round 2、収束

`CODEX VERDICT: LGTM` / `FINDINGS COMPLETE` あり / 新規指摘なし。head `48e34a21`、以降 push なし、CI **44/44 SUCCESS**。

**Codex が独自に検証した点**: 「`SwipeElement` の top-level フィールドは全て optional」— これが `isSwipeElements` の narrowing が健全である根拠で、私の主張の裏が取れました。先方の但し書き「ネストした型には必須フィールドがあり得るが、実経路では zod が守る」も正確です。

round 1 で意図的に残した 2 件の差分（`[null]` / `["str"]` が throw やゴミではなくフォールバックする）も、そのままで良いと確認されています。

File: src/utils/image_plugins/html_tailwind_markup.ts (hunk 1 above)
Comment (isamu) [reply src/utils/image_plugins/html_tailwind_markup.ts]: 正しい指摘で、**`48e34a21` で対応済み**です（round 2 でも LGTM を確認しています）。

`Array.isArray` をガードの中に取り込みました:

```ts
export const isSwipeElements = (value: unknown): value is SwipeElement[] =>
  Array.isArray(value) && value.every((entry) => typeof entry === "object" && entry !== null);
```

呼び出し側も `isSwipeElements(image.elements) && image.elements.length > 0` の順に変え、**非配列が `.every` に到達しない**ようにしています。ご指摘の `{ length: 1 }` もこれでフォールバックします。

**指摘は私が「攻撃してくれ」と書いた箇所そのものでした** — 共有モジュールへ移す際に旧コードのチェックを落とし、生成器が配列しか作っていなかったので「差分 0」と出ていました。

再測定（17ケース、throw も結果として比較）で **2 件差分**が残ります:

| 入力 | main | 本PR |
|---|---|---|
| `elements: [null]` | **throw** | html にフォールバック |
| `elements: ["str"]` | 文字列からゴミ markup | html にフォールバック |

5形すべて zod が parse で弾くので実経路からは到達せず、到達する場合は新挙動の方が良いため意図的に採用し、テストで固定しました。
